feat(lytfs): allow use of env var to auth requests

l-zeuch opened the pull request:
This is a combination of two commits:
- chore(deps): upgrade google/go-github to v55
- feat(lytfs): allow use of env var to auth requests

chore(deps:) upgrade google/go-github to v55

  Upgrade google/go-github to v55 (latest), so that the WithAuthToken
  configurator becomes available for us to use.
  
  Additionally this will get rid of the transient dependency on
  "golang.org/x/auth/oauth2", but that is tangential.
  
  Signed-off-By: Luca Zeuch <l-zeuch@email.de>

feat(lytfs): allow use of env var to auth requests

  Allow passing along an environment variable to authenticate requests to
  the GitHub API so that clients have a higher ratelimit.
  
  This is useful for developers wanting to use yagfuncdata, as often
  enough it is just not possible to do so without rapid testing.
  
  Closes #1
  
  Signed-off-by Luca Zeuch <l-zeuch@email.de>

l-zeuch committed:
chore(deps): uprade google/go-github to v55

Upgrade google/go-github to v55 (latest), so that the WithAuthToken
configurator becomes available for us to use.

Additionally this will get rid of the transient dependency on
"golang.org/x/auth/oauth2", but that is tangential.

Signed-off-by: Luca Zeuch <l-zeuch@email.de>

l-zeuch committed:
feat(lytfs): allow use of env var to auth requests

Allow passing along an environment variable to authenticate requests to
the GitHub API so that clients have a higher ratelimit.

This is useful for developers wanting to use yagfuncdata, as often
enough it is just not possible to do so without rapid testing.

Closes #1

Signed-off-by: Luca Zeuch <l-zeuch@email.de>

l-zeuch commented on cmd/lytfs/main.go:
```diff
@@ -17,7 +17,9 @@ var (
-usage: lytfs [-timeout duration]`)
+usage: lytfs [-timeout duration]
+
+To authenticate your requests, pass a GitHub personal access token via the YAGFUNCDATA_AUTH_TOKEN environment variable.`)
```
This phrasing is obviously up for debate, but it gets the point across.

jo3-l reviewed (changes requested):
Having `NewGithubClient` change behavior depending on the presence of an environment variable is surprising IMO; that should be done by the program that uses `yagfuncdata`. So WDYT about leaving `DefaultFileContentProvider` as is and instead changing `lytfs` to call `NewGitHubFileProvider` with an authenticated GitHub client if needed?

jo3-l commented on cmd/lytfs/main.go:
```diff
@@ -28,7 +31,13 @@ func main() {
 	defer cancel()
 
-	sources := yagfuncdata.DefaultSources(yagfuncdata.DefaultFileContentProvider)
+	var sources []yagfuncdata.Source
+	if token := os.Getenv("YAGFUNCDATA_AUTH_TOKEN"); token != "" {
+		fcp := yagfuncdata.NewGitHubFileProvider(github.NewClient(nil).WithAuthToken(token), "botlabs-gg", "yagpdb", "master")
+		sources = yagfuncdata.DefaultSources(fcp)
+	} else {
+		sources = yagfuncdata.DefaultSources(yagfuncdata.DefaultFileContentProvider)
+	}
```
This is a nit, but I would rather write
```suggestion
	fcp := yagfuncdata.DefaultFileContentProvider
	if token := os.Getenv("YAGFUNCDATA_AUTH_TOKEN"); token != "" {
		fcp = yagfuncdata.NewGitHubFileProvider(github.NewClient(nil).WithAuthToken(token), "botlabs-gg", "yagpdb", "master")
	}
	sources := yagfuncdata.DefaultSources(fcp)
```

l-zeuch replied:
Sure, that reads cleaner anyway.

l-zeuch committed:
format(lytfs): implement suggestion

Signed-off-by: Luca Zeuch <l-zeuch@email.de>

Co-authored-by: Joe L. <56809242+jo3-l@users.noreply.github.com>

l-zeuch committed:
style(lytfs): rename env var to LYTFS_GITHUB_TOKEN

Rename the env var used by lytfs to LYTFS_GITHUB_TOKEN to adhere to the
convention that env vars are generally prefixed by the program name that
uses them.

Formerly this was yagfuncdata, but after review this changed to lytfs.
Furthermore this token is exclusively for GitHub, thus it should also be
named GITHUB_TOKEN instead of AUTH_TOKEN.

Signed-off-by: Luca Zeuch <l-zeuch@email.de>

jo3-l reviewed (approved):
LGTM, thanks for your patience :-)

l-zeuch commented:
Thanks, no worries :D
